example/web-crypto.js

```diff
@@ -0,0 +1,254 @@
+// @param returnedFiles: Promise<{ type: 'text'|'uri', data: string, mimetype: string }[]
+function insertIntoCell(returnedFiles) {
+  const headerIds = Array.from(document.getElementById('headers').children).map(
+    (child) => child.id,
+  );
+
+  returnedFiles.forEach(async (returnedFile, i) => {
+    const returnedFileObj = await returnedFile;
+    Object.entries(returnedFileObj).forEach(([key, value]) => {
+      if (!headerIds.includes(key)) {
+        return;
+      }
+      const cell = document.getElementById(`${i}-${key}`);
+
+      if (key === 'data' && returnedFileObj.type === 'uri') {
+        // Display media
+        const elementType = returnedFileObj.mimetype.split('/')[0];
+
+        let media;
+        if (elementType === 'image') {
+          media = document.createElement('img');
+        } else if (elementType === 'video') {
+          media = document.createElement('video');
+          media.autoplay = true;
+          media.muted = true;
+        }
+
+        media.src = value;
+
+        cell.appendChild(media);
+
+        return;
+      }
+      cell.innerText = value.slice(0, 99);
+    });
+  });
+}
+
+// Decryption logic
+const onReady = async () => {
+  const output = [];
+  const promises = files.map(async (file) => {
+    const { url, filePrefix, mimetype, decryptionOptions } = file;
+
+    // Fetch the encrypted file from S3
+    const response = await fetch(url);
+    // buffer everything to mem
+    const blob = await response.blob();
+    // convert to array buffer
+    const buff = await blob.arrayBuffer();
+
+    if (!decryptionOptions) {
+      return output.push({
+        type: 'text',
+        data: 'Skipping the plaintext stuff',
+        mimetype,
+      });
+    }
+
+    // Function to convert base64-encoded string to Uint8Array
+    const bufFromB64 = (str) =>
+      Uint8Array.from(atob(str), (c) => c.charCodeAt(0));
+
+    const keyData = bufFromB64(decryptionOptions.key);
+    const iv = bufFromB64(decryptionOptions.iv);
+    const authTag = bufFromB64(decryptionOptions.authTag);
+
+    const combined = new Uint8Array([...new Uint8Array(buff), ...authTag]);
+
+    const key = await crypto.subtle.importKey(
+      'raw',
+      keyData,
+      'AES-GCM',
+      false,
+      ['decrypt'],
+    );
+
+    const plaintext = await crypto.subtle.decrypt(
+      {
+        name: 'AES-GCM',
+        iv,
+      },
+      key,
+      combined,
+    );
+
+    // Return plaintext
+    if (isViewableText(mimetype)) {
+      const enc = new TextDecoder('utf-8');
+      const decodedPlaintext = enc.decode(plaintext);
+      return { type: 'text', data: decodedPlaintext, mimetype };
+    }
+
+    // Return URL
+    const decryptedBlob = new Blob([plaintext], { type: mimetype });
+    const decryptedUrl = URL.createObjectURL(decryptedBlob);
+    return { type: 'uri', data: decryptedUrl, mimetype };
+  });
+
+  try {
+    insertIntoCell(promises);
+  } catch (err) {
+    console.error(err);
+  }
+};
```

Comment:
This is the "implementation". The rest of this PR is visual/extraneous

Reply:
Thanks, this is very readable!